### PR TITLE
Openjdk riscv trampolines

### PR DESCRIPTION
**Pull request by zakkak**

### CODE REVIEW

**Review by gigiblender — APPROVED** [review]

**Comment by gigiblender** [comment]
If T1X code is larger than 1MB, is it going to try compilation with C1X ? What happens if C1X code is larger than 1MB as well?

**Comment by zakkak** [comment]
> If T1X code is larger than 1MB, is it going to try compilation with C1X ?

Yes

> What happens if C1X code is larger than 1MB as well?

Good point. I guess we need to add a similar check and crash in that case (just in case)

**Comment by zakkak** [comment]
Bump

**Comment by zakkak** [comment]
> Good point. I guess we need to add a similar check and crash in that case (just in case)

@bakaliosdim do you think you can take care of that?

File: com.sun.c1x/src/com/sun/c1x/C1XCompiler.java [hunk 1]
```diff
@@ -114,6 +115,9 @@ public CiResult compileMethod(RiResolvedMethod method, int osrBCI, CiStatistics
                 TTY.println(String.format("C1X %4d %-70s %-45s %-50s | %3d.%dms %5dB", index, "", "", "", time / 10, time % 10, result.targetMethod().targetCodeSize()));
             }
         }
+        if (target.arch.isRISCV64() && result.targetMethod().targetCodeSize() >= RISCV64.MAX_DIRECT_JUMP_SIZE) {
+            throw new CiBailout("methods larger than 1MB are not supported in RISCV64: ");
```
Comment (zakkak) [inline com.sun.c1x/src/com/sun/c1x/C1XCompiler.java]:
```suggestion
            throw new CiBailout("methods larger than 1MB are not supported in RISCV64: "  + result.targetMethod());
```
 
Does this work? :point_up:

**Comment by zakkak** [comment]
I think this is good to go!
@kotselidis what do you thing?

**Review by timhartley — APPROVED** [review]
Looks good to me.

Many apologies for the delayed response.

**Comment by zakkak** [comment]
Bump

**Comment by zakkak** [comment]
If there are no objections till next Friday I am going to merge this :)

**Review by kotselidis — APPROVED** [review]

**Pull request merged**